core/src/main/scala/org/bitcoins/core/protocol/ln/LnTaggedFields.scala

```diff
@@ -94,31 +96,27 @@ object LnTaggedFields {
     fallbackAddress: Option[LnTag.FallbackAddressTag] = None,
     routingInfo: Option[LnTag.RoutingInfo] = None): LnTaggedFields = {
 
-    if (descriptionOrHash.isLeft) {
-      InvoiceTagImpl(
-        paymentHash = paymentHash,
-        description = descriptionOrHash.left.toOption,
-        nodeId = nodeId,
-        descriptionHash = None,
-        expiryTime = expiryTime,
-        cltvExpiry = cltvExpiry,
-        fallbackAddress = fallbackAddress,
-        routingInfo = routingInfo)
-    } else {
-
-      InvoiceTagImpl(
-        paymentHash = paymentHash,
-        description = None,
-        nodeId = nodeId,
-        descriptionHash = descriptionOrHash.right.toOption,
-        expiryTime = expiryTime,
-        cltvExpiry = cltvExpiry,
-        fallbackAddress = fallbackAddress,
-        routingInfo = routingInfo)
+    val (description, descriptionHash): (Option[LnTag.DescriptionTag],
```

Comment:
Probably would want to use `Either[LnTag.DescriptionTag, LnTag.DescriptionHashTag]` since I believe BOLT11 says that one of the two has to be defined

Reply:
That's what's being used in the signature of the containing method. My change here is to just unwrap the `Either` into a tuple, to avoid calling `InvoiceTagImpl()` twice with almost identical arguments.